### PR TITLE
[event v2] refactor coin events requested by indexer team

### PR DESCRIPTION
**Pull request by lightmark**
## Description
We haven't turned on the feature but yesterday @rtso really wants coin events v2 to get rid of `phantom` type in the name for indexing. I agree this is a legit asking.

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [X] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/coin.move [hunk 1]
```diff
@@ -167,30 +165,50 @@ module aptos_framework::coin {
...
+    struct DepositEvent has drop, store {
+        amount: u64,
+    }
+
+    /// Event emitted when some amount of a coin is withdrawn from an account.
+    struct WithdrawEvent has drop, store {
+        amount: u64,
+    }
+
+
```
Comment (rtso) [inline aptos-move/framework/aptos-framework/sources/coin.move:211]: Why do we need these?

Comment (lightmark) [reply aptos-move/framework/aptos-framework/sources/coin.move:211]: please check. they are moved here but not new code...

**Review by rtso — APPROVED** [review]
lgtm from indexer side

**Review by bowenyang007 — APPROVED** [review]
lgtm

**Pull request merged**